network/p2p/network.go

```diff
@@ -117,12 +119,47 @@ func (n *Network) Disconnected(_ context.Context, nodeID ids.NodeID) error {
 	return nil
 }
 
-// NewAppProtocol reserves an identifier for an application protocol handler and
-// returns a Client that can be used to send messages for the corresponding
-// protocol.
-func (n *Network) NewAppProtocol(handlerID uint64, handler Handler, options ...ClientOption) (*Client, error) {
-	if err := n.router.addHandler(handlerID, handler); err != nil {
-		return nil, err
+// NewClient returns a Client that can be used to send messages for the
+// corresponding protocol.
+func (n *Network) NewClient(handlerID uint64, options ...ClientOption) (*Client, error) {
+	appRequestFailedTime, err := metric.NewAverager(
+		n.namespace,
+		fmt.Sprintf("handler_%d_app_request_failed", handlerID),
```

Comment:
We could change this metric name (since this never hit the handler to begin with, it always just measures the time for the client callback to finish running).

Reply:
Without the handler dependence, I'd move NewClient to client.go file

Reply:
I don't have a strong preference for moving vs against it... but I feel like this looks better as-is since the receiver is on `Network`.